keystrokes/src/components/Character/CharacterDetails/KeybindEditor.js

```diff
@@ -49,23 +49,31 @@ function KeybindEditor({
 				spec,
 				keyBinding
 			)}`;
-			const snapShot = await ref.child(path).once('value');
-			if (
-				!Object.keys(allKeybindings).includes(
-					characterKeybindings(character, spec, keyBinding)
-				)
-			) {
-				const keybindingsWeGot = snapShot.exists()
-					? snapShot.val()
-					: [];
-				const key = characterKeybindings(character, spec, keyBinding);
-				setAllKeybindings({
-					...allKeybindings,
-					[key]: keybindingsWeGot,
-				});
+			try {
+				const snapShot = await ref.child(path).once('value');
+				if (
+					!Object.keys(allKeybindings).includes(
+						characterKeybindings(character, spec, keyBinding)
+					)
+				) {
+					const keybindingsWeGot = snapShot.exists()
+						? snapShot.val()
+						: [];
+					const key = characterKeybindings(
+						character,
+						spec,
+						keyBinding
+					);
+					setAllKeybindings({
```

Comment:
This is a bit tricky to breakdown since i don't know the model that well but seems fine. depending on the structure you might want to consider using immutability helper here